docs/concepts/registries.md

```diff
@@ -1,200 +1,232 @@
+A `RegistryOps` can be created via `RegistryOps.create(JsonElement.INSTANCE, RegistryAccess.builtinCopy())`. `RegistryAccess.builtinCopy()` creates a set of writable datapack registries, which is necessary for datagenning unregistered objects. All data generation done in a `GatherDataEvent` handler must use the same `RegistryAccess` and `RegistryOps` instances, otherwise obscure errors will occur.
 
-:::note
-Datapack Registries cannot be created with `DeferredRegister`. They can only be created through the event.
-:::
+### `Holder`s
 
-### With DeferredRegister
+As mentioned before, (normal) registries rely on `DeferredHolder`s, which are a special kind of `Holder`. A `Holder` vaguely resembles a `Pair<K, V>` that either starts with a key and has a value bound later, or starts with a value and may have a key bound later. Datapack registries extensively rely on (non-deferred) `Holder`s to reference registry elements of other registries. For example, `Biome`s refer to `Holder<PlacedFeature>`s, and `PlacedFeature`s refer to `Holder<ConfiguredFeature>`s.
 
-The `DeferredRegister` method is once again another wrapper around the above event. Once a `DeferredRegister` is created in a constant field using the `#create` overload which takes in the registry name and the mod id, the registry can be constructed via `DeferredRegister#makeRegistry`. This takes in  a supplied `RegistryBuilder` containing any additional configurations. The method already populates `#setName` by default. Since this method can be returned at any time, a supplied version of an `IForgeRegistry` is returned instead. Getting the custom registry from the supplier before `NewRegistryEvent` is fired will result in a `null` value.
+During data generation, we can use `RegistryOps#registry` to get a registry, and `Registry#getOrCreateHolderOrThrow()` to produce key-only reference holders (we only need the key in this case, since holder codecs only encode keys when using a `RegistryOps` in order to prevent circular dependencies).
 
-:::caution
-`DeferredRegister#makeRegistry` must be called before the `DeferredRegister` is added to the mod event bus via `#register`. `#makeRegistry` also uses the `#register` method to create the registry during `NewRegistryEvent`.
-:::
-
-Handling Missing Entries
-------------------------
+### `JsonCodecProvider`
 
-There are cases where certain registry objects will cease to exist whenever a mod is updated or, more likely, removed. It is possible to specify actions to handle the missing mapping through the third of the registry events: `MissingMappingsEvent`. Within this event, a list of missing mappings can be obtained either by `#getMappings` given a registry key and mod id or all mappings via `#getAllMappings` given a registry key.
+NeoForge provides a data provider for datapack registry elements that, given a registry key and a map of objects to generate, generates all JSON files for the objects in the map.
```

Comment:
The correct way of generating dp entries is through the use of `Bootstrap` and the registry access CF. Check the neo test mod for damage types as a reference.

Reply:
https://github.com/neoforged/NeoForge/blob/5e1e9fad1f16b73d716b5d51a6a653594150c987/tests/src/main/java/net/neoforged/neoforge/debug/damagesource/DamageTypeTests.java#L80-L89

Reply:
We already have docs for this actually: https://docs.neoforged.net/docs/datagen/server/datapackregistries  
Though they may also need a bit of updating, and it has a broken admonition.